api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PersonAttributeTranslatorImpl.java

```diff
@@ -0,0 +1,198 @@
+	}
+	
+	private void setExtensionValue(Extension extension, PersonAttribute personAttribute) {
+		String format = personAttribute.getAttributeType().getFormat();
+		String value = personAttribute.getValue();
+		
+		try {
+			switch (format) {
+				case JAVA_STRING_FORMAT:
+					extension.setValue(new StringType(value));
+					break;
+				case JAVA_BOOLEAN_FORMAT:
+					extension.setValue(new BooleanType(value));
+					break;
+				case OPENMRS_LOCATION_FORMAT:
+					handleLocationFormat(extension, value);
+					break;
+				case OPENMRS_CONCEPT_FORMAT:
+					handleConceptFormat(extension, value);
+					break;
+				default:
+					throw new UnsupportedOperationException("Unsupported attribute format: " + format);
+			}
+		}
+		catch (Exception e) {
+			throw new IllegalStateException("Error processing attribute value for format: " + format, e);
+		}
+	}
+	
+	private void handleLocationFormat(Extension extension, String locationId) {
+		if (locationId == null) {
+			return;
+		}
+		
+		Reference locationReference = new Reference().setType(LOCATION_TYPE).setReference(LOCATION_TYPE + "/" + locationId);
+		
+		Optional.ofNullable(locationService.getLocation(locationId))
+		        .ifPresent(location -> locationReference.setDisplay(location.getDisplayString()));
+		
+		extension.setValue(locationReference);
+	}
+	
+	private void handleConceptFormat(Extension extension, String conceptId) {
+		if (conceptId == null) {
+			return;
+		}
+		
+		Concept concept = conceptService.getConcept(conceptId);
+		if (concept != null) {
+			CodeableConcept codeableConcept = conceptTranslator.toFhirResource(concept);
+			extension.setValue(codeableConcept);
+		}
+		
+	}
+	
+	private boolean isValidPatientAttributeExtension(Extension extension) {
+		return extension != null && extension.hasUrl()
+		        && extension.getUrl().equals(FhirConstants.OPENMRS_FHIR_EXT_PERSON_ATTRIBUTE) && extension.hasExtension()
+		        && extension.getExtensionFirstRep().hasValue();
+	}
+	
+	private String extractAttributeTypeName(String url) {
+		return url.substring(url.lastIndexOf("#") + 1);
+	}
+	
+	private PersonAttributeType getPersonAttributeType(String typeName) {
+		PersonAttributeType attributeType = personService.getPersonAttributeTypeByName(typeName);
+		if (attributeType == null) {
+			throw new IllegalStateException("PersonAttributeType not found: " + typeName);
+		}
+		return attributeType;
+	}
+	
+	private void setPersonAttributeValue(PersonAttribute personAttribute, Type extensionValue) {
+		try {
+			if (extensionValue instanceof BooleanType) {
+				personAttribute.setValue(((BooleanType) extensionValue).getValueAsString());
+			} else if (extensionValue instanceof StringType) {
+				personAttribute.setValue(((StringType) extensionValue).getValue());
+			} else if (extensionValue instanceof Reference) {
+				String reference = ((Reference) extensionValue).getReference();
+				personAttribute.setValue(reference.substring(reference.lastIndexOf("/") + 1));
```

Comment:
Use `FhirUtils#referenceToId()` or `ReferenceHandlingTranslator#getReferenceId()`. There are quite a few edge cases where this won't work.